compiler/luci/pass/src/SubstituteSqueezeToReshapePass.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/SubstituteSqueezeToReshapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+/**
+ * @brief Return TRUE if all dim is known && dim > 0 && at least dim 1 exist
+ */
+bool can_squeeze_shape(luci::CircleNode *node)
+{
+  int dim_1_cnt = 0;
+  for (uint32_t r = 0; r < node->rank(); ++r)
+  {
+    if (not node->dim(r).known())
+      return false;
+    if (node->dim(r).value() < 1)
+      return false;
+    if (node->dim(r).value() == 1)
+      dim_1_cnt++;
+  }
+  return dim_1_cnt > 0;
+}
+
+/**
+ * @brief return shape vector of input excluding squeeze_dims
+ */
+std::vector<uint32_t> new_shape_with_squeeze(const luci::CircleNode *input,
+                                             const std::vector<int32_t> &squeeze_dims)
+{
+  std::vector<uint32_t> new_shape;
+
+  if (squeeze_dims.size() == 0)
+  {
+    // prepare new_shape from input shape excluding dim 1
+    for (uint32_t r = 0; r < input->rank(); ++r)
+    {
+      auto v = input->dim(r).value();
+      if (v != 1)
+        new_shape.push_back(v);
+    }
+  }
+  else
+  {
+    // prepare new_shape from input shape excluding squeeze_dims
+    std::vector<uint32_t> temp_shape;
+    for (uint32_t r = 0; r < input->rank(); ++r)
+      temp_shape.push_back(input->dim(r).value());
+    for (auto dim : squeeze_dims)
+    {
+      assert(temp_shape.at(dim) == 1);
+      temp_shape.at(dim) = 0;
```

Comment:
Input node can include zero dimension such as [1, 0, 2, 3].
If we mark erased dimension as 0, real zero dimension cannot be distinguished.
So how about using -1 instead?

```suggestion
      temp_shape.at(dim) = -1;
```

Reply:
this block will be removed as shape for CircleReshape will be from output of CircleSqueeze node.